msteams-platform/apps-in-teams-meetings/build-apps-for-anonymous-user.md

```diff
@@ -0,0 +1,229 @@
+
+**Paged Get Members API**
+
+```csharp
+{ 
+  "members": [ 
+    { 
+      "id": "<<GUID>>", 
+      "name": "AnonTest (Guest)",  
+      "tenantId": "<<GUID>>", 
+      "userRole": "anonymous" 
+    } 
+  ] 
+} 
+```
+
+| **Property Name** | **Description** |
+| --- | --- |
+| `id` | 29:XXXX encrypted user id |
+| `name` | provided by anonymous user, when joining meeting |
+| `tenantId` | Tenant ID of the meeting organizer |
+| `userRole` | `anonymous`, represent anonymous user |
+
+> [!NOTE]
+> The ID returned in bot API payload and Teams Client SDK are not same.
+
+**Get Single Member API**
+
+```csharp
+{ 
+  "id": "<<GUID>>", 
+  "name": "AnonTest (Guest)", 
+  "tenantId": "<<GUID>>", 
+  "userRole": "anonymous" 
+} 
+```
+
+**ConversationUpdate Event – MembersAdded Payload object**
+
+```csharp
+{ 
+  "membersAdded": [ 
+    { 
+      "id": "<<GUID1>>" 
+    } 
+  ], 
+  "type": "conversationUpdate", 
+  "timestamp": "<<timestamp>>", 
+  "id": "<<GUID2>>", 
+  "channelId": "msteams", 
+  "serviceUrl": "<<serviceURL>>", 
+  "from": { 
+    "id": "<<GUID3>>" 
+  }, 
+  "conversation": { 
+    "isGroup": true, 
+    "tenantId": "<<tenant id>>", 
+    "id": "<<chat id>>" 
+  }, 
+  "recipient": { 
+    "id": "<<bot id>>", 
+    "name": "bot name" 
+  }, 
+  "channelData": { 
+    "tenant": { 
+      "id": "<<tenant id>>" 
+    }, 
+    "source": null, 
+    "meeting": { 
+      "id": "<<meeting id>>" 
+    } 
+  } 
+} 
+```
+
+> [!NOTE]
+>
+> * When an anonymous user is added to the meeting, `membersAdded` payload object doesn't have `aadObjectId` field.
+> * When an anonymous user is added to the meeting, `from` object in the payload always have the id of the meeting organizer, even if the anonymous user was added by another presenter.
+
+**ConversationUpdate Event – MembersRemoved Payload object**
+
+```csharp
+{
+  "membersRemoved": [
+    {
+      "id": "<<GUID1>>" 
+    }
+  ],
+  "type": "conversationUpdate",
+  "timestamp": "<<timestamp>>",
+  "id": "<<GUID2>>",
+  "channelId": "msteams",
+  "serviceUrl": "<<serviceURL>>",
+  "from": {
+    "id": "<<GUID3>>"
+  },
+  "conversation": {
+    "isGroup": true,
+    "tenantId": "<<tenant id>>",
+    "id": "<<chat id>>"
+  },
+  "recipient": {
+    "id": "<<bot id>>",
+    "name": "bot name"
+  },
+  "channelData": {
+    "tenant": {
+      "id": "<<tenant id>>"
+    },
+    "source": null,
+    "meeting": {
+      "id": "<<meeting id>>"
+    }
+  }
+}
+```
+
+> [!NOTE]
+>
+> * When an anonymous user is removed from a meeting, `membersRemoved` payload object doesn't have `aadObjectId` field.
+> * When an anonymous user is removed from a meeting, `from` object in the payload always have the id of the meeting organizer, even if the anonymous user was removed by another presenter.
+
+**Create Conversation API**
+
+Create conversation API is not supported for anonymous users. The anonymous user receives the following error payload:
```

Comment:
It's the bot that receives this error, right, not the anonymous user.

"If a bot attempts to create a conversation with an anonymous user, it will receive a 400 Bad Request status code, with the following payload."